### PR TITLE
drivers: ipm: remove nxp,imx-mu-rev2 compatible

### PR DESCRIPTION
**Pull request by danieldegrasse**
Remove `nxp,imx-mu-rev2` compatible. The `nxp,imx-mu` compatible already handles the same hardware, so we can simply use `CONFIG_HAS_MCUX` to detect when we need to use legacy i.MX HAL APIs, versus when we can use MCUX HAL API for the messaging unit.

### CODE REVIEW

**Review by dcpleung — DISMISSED** [review]

File: drivers/ipm/CMakeLists.txt [hunk 1]
```diff
@@ -6,7 +6,6 @@ zephyr_library()
 
 zephyr_library_sources_ifdef(CONFIG_IPM_MCUX   ipm_mcux.c)
```
Comment (iuliana-prodan) [inline drivers/ipm/CMakeLists.txt:7]: I think there's a typo in commit message here: "The
driver can now be enabled with _CONFIG_IPM_IMX_REV2_." - s/CONFIG_IPM_IMX_REV2/_CONFIG_IPM_IMX

Comment (danieldegrasse) [reply drivers/ipm/CMakeLists.txt:7]: There is, thanks for noticing. I've updated the PR

**Review by iuliana-prodan — DISMISSED** [review]

**Review by dleach02 — DISMISSED** [review]

**Comment by mmahadevan108** [comment]
@danieldegrasse , please update `west.yml`

**Comment by danieldegrasse** [comment]
> @danieldegrasse , please update west.yml

Done, thanks

**Review by mmahadevan108 — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Review by iuliana-prodan — APPROVED** [review]

**Pull request merged**